STModCellular.cpp

```diff
@@ -73,37 +124,50 @@ nsapi_error_t STModCellular::soft_power_on() {
         _at->write_int(2);
         _at->write_int(2);
         _at->cmd_stop_read_resp();
-
         err = _at->get_last_error();
+        _at->unlock();
+
         if (err == NSAPI_ERROR_OK) {
             tr_debug("Flow control turned ON");
         } else {
             tr_error("Failed to enable hw flow control");
         }
-
-        _at->restore_at_timeout();
```

Comment:
Moving the `unlock()` about the `if` statement seems fine for me but I think we still need to `restore_at_timeout()` before releasing the lock.

Reply:
Right - Done